model/histogram/float_histogram.go

```diff
@@ -410,53 +416,7 @@ func (h *FloatHistogram) Add(other *FloatHistogram) (*FloatHistogram, error) {
 
 // Sub works like Add but subtracts the other histogram.
 func (h *FloatHistogram) Sub(other *FloatHistogram) (*FloatHistogram, error) {
-	if h.UsesCustomBuckets() != other.UsesCustomBuckets() {
-		return nil, ErrHistogramsIncompatibleSchema
-	}
-	if h.UsesCustomBuckets() && !FloatBucketsMatch(h.CustomValues, other.CustomValues) {
-		return nil, ErrHistogramsIncompatibleBounds
-	}
-
-	if !h.UsesCustomBuckets() {
-		otherZeroCount := h.reconcileZeroBuckets(other)
-		h.ZeroCount -= otherZeroCount
-	}
-	h.Count -= other.Count
-	h.Sum -= other.Sum
-
-	var (
-		hPositiveSpans       = h.PositiveSpans
-		hPositiveBuckets     = h.PositiveBuckets
-		otherPositiveSpans   = other.PositiveSpans
-		otherPositiveBuckets = other.PositiveBuckets
-	)
-
-	if h.UsesCustomBuckets() {
-		h.PositiveSpans, h.PositiveBuckets = addBuckets(h.Schema, h.ZeroThreshold, true, hPositiveSpans, hPositiveBuckets, otherPositiveSpans, otherPositiveBuckets)
-		return h, nil
-	}
-
-	var (
-		hNegativeSpans       = h.NegativeSpans
-		hNegativeBuckets     = h.NegativeBuckets
-		otherNegativeSpans   = other.NegativeSpans
-		otherNegativeBuckets = other.NegativeBuckets
-	)
-
-	switch {
-	case other.Schema < h.Schema:
-		hPositiveSpans, hPositiveBuckets = reduceResolution(hPositiveSpans, hPositiveBuckets, h.Schema, other.Schema, false, true)
-		hNegativeSpans, hNegativeBuckets = reduceResolution(hNegativeSpans, hNegativeBuckets, h.Schema, other.Schema, false, true)
-		h.Schema = other.Schema
-	case other.Schema > h.Schema:
-		otherPositiveSpans, otherPositiveBuckets = reduceResolution(otherPositiveSpans, otherPositiveBuckets, other.Schema, h.Schema, false, false)
-		otherNegativeSpans, otherNegativeBuckets = reduceResolution(otherNegativeSpans, otherNegativeBuckets, other.Schema, h.Schema, false, false)
-	}
-
-	h.PositiveSpans, h.PositiveBuckets = addBuckets(h.Schema, h.ZeroThreshold, true, hPositiveSpans, hPositiveBuckets, otherPositiveSpans, otherPositiveBuckets)
-	h.NegativeSpans, h.NegativeBuckets = addBuckets(h.Schema, h.ZeroThreshold, true, hNegativeSpans, hNegativeBuckets, otherNegativeSpans, otherNegativeBuckets)
-
-	return h, nil
+	return h.Add(other.Copy().Mul(-1))
```

Comment:
Hmm, isn't this much less efficient? We're creating one more copy. 
Let's move this to a different PR and not do it here.